modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaConnectionProviderModelParser.java

```diff
@@ -141,23 +152,69 @@ public boolean isExcludedFromConnectivitySchema() {
   @Override
   public Optional<OAuthModelProperty> getOAuthModelProperty() {
     List<OAuthGrantType> grantTypes = new LinkedList<>();
-    element.getAnnotation(AuthorizationCode.class)
-        .ifPresent(a -> grantTypes.add(new AuthorizationCodeGrantType(a.accessTokenUrl(),
-                                                                      a.authorizationUrl(),
-                                                                      a.accessTokenExpr(),
-                                                                      a.expirationExpr(),
-                                                                      a.refreshTokenExpr(),
-                                                                      a.defaultScopes(),
-                                                                      a.credentialsPlacement(),
-                                                                      a.includeRedirectUriInRefreshTokenRequest())));
-
-    element.getAnnotation(ClientCredentials.class).ifPresent(a -> grantTypes.add(new ClientCredentialsGrantType(a.tokenUrl(),
-                                                                                                                a.accessTokenExpr(),
-                                                                                                                a.expirationExpr(),
-                                                                                                                a.defaultScopes(),
-                                                                                                                a.credentialsPlacement()))
-
-    );
+    mapReduceSingleAnnotation(element, CONNECTION_PROVIDER_NAME, element.getName(), AuthorizationCode.class,
```

Comment:
would it be possible to add a couple methods for creating the AuthorizationCodeGrantType and the ClientCredentialsGrantType in order to make the getOAuthModelProperty() method more modular/readable? what do you think?

Reply:
Totally, I will do it.